File: KOTHA_MADHUVANI/Arrays and Strings/problem14.java
public class problem14 {
    public static int findMaxConsecutiveOnes(int[] nums) {
        int maxCount = 0;
        int currentCount = 0;

        for (int num : nums) {
            if (num == 1) {
                currentCount++;
            } else {
                maxCount = Math.max(maxCount, currentCount);
                currentCount = 0;
            }
        }

        return Math.max(maxCount, currentCount);
    }

    public static void main(String[] args) {
        int[] nums = { 1, 1, 0, 1, 1, 1 };
        int maxConsecutiveOnes = findMaxConsecutiveOnes(nums);
        System.out.println(maxConsecutiveOnes); // Output: 3
    }
}

File: Assignment 1/Problem14.py
def findMaxConsecutiveOnes(nums):
    max_ones = 0
    current_ones = 0

    for num in nums:
        if num == 1:
            current_ones += 1
            max_ones = max(max_ones, current_ones)
        else:
            current_ones = 0

    return max_ones

nums = [1, 1, 0, 1, 1, 1]
max_consecutive_ones = findMaxConsecutiveOnes(nums)
print(max_consecutive_ones)
